src/player.ts

```diff
@@ -315,13 +315,17 @@ export default class Player extends EventEmitter {
     // Add to nearby federation
     if (building !== Building.GaiaFormer && !hex.belongsToFederationOf(this.player)) {
       const group: GaiaHex[] = this.buildingGroup(hex);
+      const sats: GaiaHex[] = this.satelittesNearby(map, hex);
       const hasFederation = group.some(hx => hx.belongsToFederationOf(this.player));
```

Comment:
The only change needed is here: 

```
const hasFederation = group.concat(map.grid.neighbours(hex)).some(hx => hx.belongsToFederationOf(this.player));
```

And that's it. No other line necessary. Just this one line :)

With the current code, a new mine next to a satellite will get added to the federation, but any other building next to it (but far away from the satellite) won't.

Reply:
oh yes, much cleaner.
BTW, isn't the current code doing what you are saying anyway, given that the group code is the same?

Reply:
The problem is that you don't set `hasFederation` to `true` if a satellite has the federation. 

So this:

```
      if (hasFederation) {
        // Add federation to ALL buildings of the group, not just the new buidling
        for (const h of group) {
          h.addToFederationOf(this.player);
        }
      }
```

is not triggered!